astropy/coordinates/baseframe.py

```diff
@@ -156,6 +156,69 @@ def __new__(mcls, name, bases, members):
                 'Could not find all expected BaseCoordinateFrame class '
                 'attributes.  Are you mis-using FrameMeta?')
 
+        # Unless overridden, velocity name defaults are:
+        #   * pm_{lon}_cos{lat} for SphericalCosLatDifferential
+        #   * radial_velocity for any d_distance, including RadialDifferential
+        #   * v_{x,y,z} for d_{x,y,z} for CartesianDifferential
+        # where {lon} and {lat} are the names of the angular components
+        if repr_info is None:
+            repr_info = dict()
+
+        if r.SphericalRepresentation in repr_info:
+            sph_mappings = repr_info[r.SphericalRepresentation]
+            component_map = dict([(m.reprname, m.framename)
+                                  for m in sph_mappings])
+
+        else:
+            component_map = dict()
+
+        component_map.setdefault('lon', 'lon')
+        component_map.setdefault('lat', 'lat')
+
+        if r.SphericalCosLatDifferential not in repr_info:
+            repr_info[r.SphericalCosLatDifferential] = [
+                RepresentationMapping(
+                    'd_lon_coslat',
+                    'pm_{lon}_cos{lat}'.format(**component_map),
+                    u.mas/u.yr),
+                RepresentationMapping('d_lat',
+                                      'pm_{lat}'.format(**component_map),
+                                      u.mas/u.yr),
+                RepresentationMapping('d_distance', 'radial_velocity',
+                                      u.km/u.s)
+            ]
+
+        if r.SphericalDifferential not in repr_info:
+            repr_info[r.SphericalDifferential] = [
+                RepresentationMapping('d_lon',
+                                      'pm_{lon}'.format(**component_map),
+                                      u.mas/u.yr),
+                RepresentationMapping('d_lat',
+                                      'pm_{lat}'.format(**component_map),
+                                      u.mas/u.yr),
+                RepresentationMapping('d_distance', 'radial_velocity',
+                                      u.km/u.s)
+            ]
+
+        if r.CartesianDifferential not in repr_info:
+            repr_info[r.CartesianDifferential] = [
+                RepresentationMapping('d_x', 'v_x', u.km/u.s),
+                RepresentationMapping('d_y', 'v_y', u.km/u.s),
+                RepresentationMapping('d_z', 'v_z', u.km/u.s)]
+
+        # Unit* classes should follow the same naming conventions
+        if r.SphericalRepresentation in repr_info:
```

Comment:
Why do we need the `if` here? You have set this entry above, no? (same further down)